workers/unity/Packages/com.improbable.gdk.playerlifecycle/Systems/PlayerCreation/SendCreatePlayerRequestSystem.cs

```diff
@@ -11,16 +14,26 @@ namespace Improbable.Gdk.PlayerLifecycle
     [UpdateInGroup(typeof(SpatialOSUpdateGroup))]
     public class SendCreatePlayerRequestSystem : ComponentSystem
     {
-        private readonly EntityId playerCreatorEntityId = new EntityId(1);
-
         private CommandSystem commandSystem;
         private WorkerSystem workerSystem;
         private ILogDispatcher logDispatcher;
 
-        private ComponentGroup initializationGroup;
-
         private byte[] serializedArgumentsCache;
 
+        private bool playerRequestQueued;
+        private int playerCreationAttempts = 0;
+
+        private int playerCreatorQueryAttempts = 0;
+        private long? playerCreatorQueryId;
+
+        private readonly List<EntityId> playerCreatorEntityIds = new List<EntityId>();
+
+        private readonly EntityQuery playerCreatorQuery = new EntityQuery
+        {
+            Constraint = new ComponentConstraint(PlayerCreator.ComponentId),
+            ResultType = new SnapshotResultType()
```

Comment:
Isn't the other result type more appropriate? We don't care about the data within the entity - just the ID

Reply:
Using the `CountResultType` only returns the number of entities that match the query

to get the entity IDs we have to get the full dictionary of results and ignore the values

Reply:
That's annoying 😕